lib/ghomfp.gi

```diff
@@ -1229,6 +1230,44 @@ local F,str;
     GroupHomomorphismByImagesNC(F,G,GeneratorsOfGroup(F),GeneratorsOfGroup(G));
 end);
 
+InstallGlobalFunction(ProcessEpimorphismToNewFpGroup,
+function(hom)
+local s,r,fam,fas,fpf,mapi;
+  if not (HasIsSurjective(hom) and IsSurjective(hom)) then
+    Info(InfoWarning,1,"ill created epimorphismfp");
+    return; # hom might be ill defined.
+  fi;
+  s:=Source(hom);
+  r:=Range(hom);
+  mapi:=MappingGeneratorsImages(hom);
+  if mapi[2]<>GeneratorsOfGroup(r) then
+    return; # the method does not apply here. One could try to deal with the
+    #extra generators separately, but that is too much work for what is
+    #intended as a minor hint.
+    Error("fp generators?");
+  fi;
+  s:=SubgroupNC(s,mapi[1]);
+  fam:=FamilyObj(One(r));
+  fas:=FamilyObj(One(s));
+  if IsPermCollection(s) or IsMatrixCollection(s) 
```

Comment:
Any particular reason you are not using `IsPermGroup(s) or IsMatrixGroup(s)` here? There should be no functional difference, but IMHO it conveys the intention of the code more naturally.

Reply:
The reason for the category test was that I was thinking in the context of `CanEasily...`. It could as well be the `Group` test.